services/bots.py

```diff
@@ -38,14 +38,17 @@ def _get_installation_weight(installation: GithubAppInstallation) -> int:
         return MAX_GITHUB_APP_SELECTION_WEIGHT
     seconds_in_hour = 3600
     age_hours = (age.seconds // seconds_in_hour) + age.days * 24
-    return age_hours + 2**age.days
+    # Prevent clock differences from making the weight negative
+    return max(1, age_hours + 2**age.days)
 
 
 def _get_apps_from_weighted_selection(
```

Comment:
I think it might be useful to add some function documentation around what it's returning because it looks like we're significantly changing the functionality. If I understand the code correctly, we're trying to get an ordered list of apps that's random based on the weighting (so that we don't _always_ use the highest weighted one). The return is no longer just a "random subset of the apps" but an ordered list of apps (with some randomness).

Reply:
That is true... good callout here.